cfn-resources/teams/cmd/resource/team-user/team_user_test.go

```diff
@@ -0,0 +1,261 @@
+//go:build unit
+
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package teamuser
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+
+	mock_util "github.com/mongodb/mongodbatlas-cloudformation-resources/test/mocks"
+)
+
+func TestInitUserSet(t *testing.T) {
+	user1 := "user1"
+	user2 := "user2"
+	user3 := "user3"
+	// Create a slice of CloudAppUser objects
+	users := []atlasv2.CloudAppUser{
+		{Id: &user1},
+		{Id: &user2},
+		{Id: &user3},
+	}
+
+	// Call the initUserSet function to create a map of user IDs
+	usersSet := initUserSet(users)
+
+	// Check that the map contains the expected user IDs
+	assert.True(t, usersSet[user1])
+	assert.True(t, usersSet[user2])
```

Comment:
this is the test assert style I'm more used to, and I'm not against it. But I think the more idiomatic Go for this is to use table style with an anonymous struct and a slice, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/util/type_conversion_test.go#L32..L69

Reply:
I agree with you but I think rewriting this in table style wouldn't change it much because the test case here is simple enough, it's just one case.  You can see below the others have been written in table style.

Resolving this one, feel free to re-open if you have a strong opinion.